packages/carbon-web-components/src/components/pagination-nav/pagination-nav.ts

```diff
@@ -0,0 +1,288 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2019, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { LitElement, property, html, css } from 'lit-element';
+import { carbonElement as customElement } from '../../globals/decorators/carbon-element';
+import settings from 'carbon-components/es/globals/js/settings';
+import styles from './pagination-nav.scss';
+import { classMap } from 'lit-html/directives/class-map';
+
+const { prefix } = settings;
+
+function calculateCuts(
+  page: number,
+  totalItems: number,
+  itemsDisplayedOnPage: number,
+) {
+  const split = (itemsDisplayedOnPage - 1) / 2;
+
+  let frontHiddenIndex = page - Math.ceil(split);
+  let backHiddenIndex = page + Math.floor(split) - 1;
+
+
+  if (frontHiddenIndex < 1) {
+    backHiddenIndex += Math.abs(frontHiddenIndex - 1);
+  }
+
+  if (backHiddenIndex > totalItems - 1) {
+    frontHiddenIndex -= (Math.abs(totalItems - backHiddenIndex - 1));
+  }
+
+  return {
+    front: frontHiddenIndex,
+    back: backHiddenIndex
+  };
+}
+
+/**
+ * Pagination Navigation.
+ *
+ * @element bx-pagination
+ */
+@customElement(`${prefix}-pagination-nav`)
+class BXPaginationNav extends LitElement {
+
+  /**
+   * The maxiumum number of items to show in pager nav.
+   *   - includes overflow items
+   *   - excludes prev/next buttons
+   *   - minumum value of 5
+   */
+  @property({ attribute: 'visible', reflect: true, type: Number})
+  visible = 1;
+
+  /**
+   * Whether to allow looping between first/last pages.
+   */
+  @property({ attribute: "loop", type: Boolean})
+  loop = false;
+
+  /**
+   * Active page index (0-based).
+   */
+  @property({ attribute: "page", reflect: true, type: Number })
+  page = 0;
```

Comment:
Being 0-based, I can see this confusing users. Thoughts on making it 1-based?

Reply:
I'm not opposed, but this matches the @carbon/react functionality

Reply:
Fair enough. +1 for consistency.